os/arch/arm/src/amebad/amebad_serial.c

```diff
@@ -819,6 +806,54 @@ int up_putc(int ch)
 	}
 
 	up_lowputc(ch);
+
+}
+
+int up_putc(int ch)
```

Comment:
This is the function of the chipset vendor. Is there no way to do this functionality on common code?

Reply:
These changes can be moved to serial.c. The reason for making the changes here is since the problem is with only rtk boards. If this method is fine, I will move these changes to serial.c.

Reply:
The serial.c does not work with printf and dbg. How would you move there?

Reply:
A new function can be declared like up_lineputc in arch.h and defined in serial.c, which is similar to the up_putc function in this PR. We can link this to lib_lowoutstream_putc (currently up_putc is linked to it).  This new function up_lineputc will use a buffer and will finally send the characters to up_putc to output once we get a whole line. Please let me know if this is fine.

Reply:
Sorry for late reply.
The serial.c has the functions of uart driver. There is no other function. Could you check where other lib_xxxstream function?
Could you write simple ppt to show how it flows?